src/overlays/actors/ovl_Bg_Hidan_Rock/z_bg_hidan_rock.c

```diff
@@ -47,31 +61,349 @@ static ColliderCylinderInit D_8088BF8C = {
+
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+    func_800F89E8(&this->unk_170);
+}
+
+void func_8088B24C(BgHidanRock* this) {
+    this->dyna.actor.flags = this->dyna.actor.flags | 0x30;
+    this->actionFunc = func_8088B990;
+}
+
+void func_8088B268(BgHidanRock* this, GlobalContext* globalCtx) {
+    static f32 D_8088BFC0 = 0.0f;
+    f32 sp2C;
+    s32 temp_v1;
+    s32 frame;
+    Player* player = PLAYER;
+
+    if (this->dyna.unk_150 != 0.0f) {
+        if (this->unk_16A == 0) {
+            if (D_8088BFC0 == 0.0f) {
+                if (this->dyna.unk_150 > 0.0f) {
+                    D_8088BFC0 += 0.01f;
+                } else {
+                    D_8088BFC0 -= 0.01f;
+                }
+            }
+
+            this->dyna.actor.speedXZ = this->dyna.actor.speedXZ + 0.05f;
+            this->dyna.actor.speedXZ = (this->dyna.actor.speedXZ > 2.0f) ? 2.0f : this->dyna.actor.speedXZ;
+
+            if (D_8088BFC0 > 0.0f) {
+                temp_v1 = Math_StepToF(&D_8088BFC0, 20.0f, this->dyna.actor.speedXZ);
+            } else {
+                temp_v1 = Math_StepToF(&D_8088BFC0, -20.0f, this->dyna.actor.speedXZ);
+            }
+
+            this->dyna.actor.world.pos.x = (Math_SinS(this->dyna.unk_158) * D_8088BFC0) + this->dyna.actor.home.pos.x;
+            this->dyna.actor.world.pos.z = (Math_CosS(this->dyna.unk_158) * D_8088BFC0) + this->dyna.actor.home.pos.z;
+
+            if (temp_v1 != 0) {
+                player->stateFlags2 &= ~0x10;
+                this->dyna.unk_150 = 0.0f;
+                this->dyna.actor.home.pos.x = this->dyna.actor.world.pos.x;
+                this->dyna.actor.home.pos.z = this->dyna.actor.world.pos.z;
+                D_8088BFC0 = 0.0f;
+                this->dyna.actor.speedXZ = 0.0f;
+                this->unk_16A = 5;
+            }
+
+            func_8002F974(&this->dyna.actor, NA_SE_EV_ROCK_SLIDE - SFX_FLAG);
+        } else {
+            player->stateFlags2 &= ~0x10;
+            this->dyna.unk_150 = 0.0f;
+            if (this->unk_16A != 0) {
+                this->unk_16A--;
+            }
+        }
+    } else {
+        this->unk_16A = 0;
+    }
+
+    sp2C = this->dyna.actor.world.pos.z - D_8088BF60.z;
+    if (sp2C < 0.5f) {
+        Flags_SetSwitch(globalCtx, this->dyna.actor.params);
+        Math_Vec3f_Copy(&this->dyna.actor.home.pos, &D_8088BF60);
+        this->dyna.actor.world.pos.x = D_8088BF60.x;
+        this->dyna.actor.world.pos.z = D_8088BF60.z;
+        this->dyna.actor.speedXZ = 0.0f;
+        D_8088BFC0 = 0.0f;
+        player->stateFlags2 &= ~0x10;
+        this->actionFunc = func_8088B79C;
+    }
+
+    frame = globalCtx->gameplayFrames & 0xFF;
+    if ((globalCtx->gameplayFrames & 0x100) != 0) {
+        this->unk_16C = 0.0f;
+    } else if (frame < 0x80) {
+        this->unk_16C = sinf(frame * (4 * 0.001f * M_PI)) * 19.625f;
+    } else if (frame < 0xE6) {
```

Comment:
```suggestion
    } else if (frame < 128) {
        this->unk_16C = sinf(frame * (4 * 0.001f * M_PI)) * 19.625f;
    } else if (frame < 230) {
```
These frame counts make more sense in decimal, I think.